docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+The point of entry for this feature will be at TimeTableCommandParser, which will decide which of the commands 
+to return through `parseTimeTableCommand()`. If the TimeTableCommand is returned and executed, the 
+TimeTableManager will carry out the associated commands, adding, deleting or viewing the Lessons in the timetable.
+
+## Feature explanation with sequence diagrams
+
+### Main loop sequence
+When ra.VI runs, there are 3 phases to its lifecycle.
+
+1. Initialisation
+2. Main command loop
+3. Saving and exiting
+
+The main sequence diagram can be broken into three parts representing each of these phases.
+
+![Sequence diagram 1 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram1.png?raw=true)
+ra.VI's `start()` method initialises all the classes used by the main class, like TextUi, InputOutputManager, and more.
+TimeTableManager may need user input to be initialised, and so it is placed in a validation loop.
+A welcome message is shown at the end of the initialisation phase.
+
+![Sequence diagram 2 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram2.png?raw=true)
+The main loop is contained in the 'runCommandLoopUntilExitCommand()' method.
+The main loop follows the following steps:
+1. Get the user command as a string.
+2. Parse the command using Parser. Parser will return a Command object.
+3. Execute the Command object with the `execute()` method. This will do the necessary work and return a CommandResult object.
+4. If any data was changed, StateManager will run `saveState()` to facilitate undo commands.
+5. Finally, use the CommandResult object to show the result of the command to the user using TextUi.
+
+Note that the Command and CommandResult objects are destroyed at the end of the method. Additionally, the Parser object
+is destroyed after each use.
+
+![Sequence diagram 3 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram3.png?raw=true)
+After exiting the main loop, the main class calls InputOutputManager's saving methods to save all user data.
+After that, the program terminates.
+
+### Add/Delete Feature
+This feature is facilitated by the TaskManager, ModuleManager classes.
+Extending from the abstract Command class are the AddModule, AddTask Command classes. This feature implements the following operations:
+* AddTask - Add a task to the task list through `TaskManager.add()`
+* AddModule - Add a module to the module list through `ModuleManager.add()`
+* DeleteTask - Deletes a task from the task list through `TaskManager.delete()`
+* DeleteModule - Deletes a module from the module list through `ModuleManager.delete()`
+
+![Sequence diagram for AddCommand](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/AddCommandSequenceDiagram.png?raw=true)
```

Comment:
Should term "construct" above constructor arrows be omitted for creation of new objects as it is implied by UML diagram notation?